ibis/backends/base/sqlglot/compiler.py

```diff
@@ -269,14 +269,24 @@ def visit_Alias(self, op, *, arg, name):
         return arg
 
     @visit_node.register(ops.Literal)
-    def visit_Literal(self, op, *, value, dtype, **kw):
+    def visit_Literal(self, op, *, value, dtype):
         if value is None:
             if dtype.nullable:
                 return NULL if dtype.is_null() else self.cast(NULL, dtype)
             raise com.UnsupportedOperationError(
                 f"Unsupported NULL for non-nullable type: {dtype!r}"
             )
-        elif dtype.is_integer():
+        else:
+            result = self.visit_NonNullLiteral(op, value=value, dtype=dtype)
+            if result is None:
```

Comment:
Is `None` used for error signaling here?

Reply:
No, it's used to indicate to use the default literal rule

Reply:
The method naming is a bit hard to understand here.

Reply:
- `visit_NonNullLiteral`: handles the non `None` case; avoids each backend having to handle `None`
- `visit_Literal`: handles `None` -> calls `visit_NonNullLiteral` -> calls `visit_DefaultLiteral` if the former returns `None`
- `visit_DefaultLiteral`: the base implementation of literal translation.

Most backends should only ever implement `visit_NonNullLiteral`

The point of this is to avoid enormous amounts of duplication of literal translation, most of which is the same across backends for the core types.

Reply:
Done.

Reply:
Added docstrings for the relevant methods.